python/lsst/pipe/tasks/colorterms.py

```diff
@@ -19,130 +19,157 @@
-        self.c2 = c2
-
-    def __str__(self):
-        return "%s %s [%g %g %g]" % (self.primary, self.secondary, self.c0, self.c1, self.c2)
-
-    @staticmethod
-    def setColorterms(colorterms, device=None):
-        """!Replace or update the cached Colorterms dict
-
-        @param[in,out] colorterms  a dict of device: Colorterm
-        @param[in] device  device name, or None;
-            if device is None then colorterms replaces the internal catched dict,
-            else the internal cached dict is updated with device: colorterms[device]
-        """
-        if device:
-            Colorterm._colorterms[device] = colorterms[device]
-        else:
-            Colorterm._colorterms = colorterms
+import fnmatch
 
-        Colorterm.setActiveDevice(device, allowRaise=False)
+import numpy as np
 
-    @staticmethod
-    def setActiveDevice(device, allowRaise=True):
-        """!Set or clear the default colour terms
+from lsst.pex.config import Config, Field, ConfigDictField
+from lsst.afw.image import Filter
 
-        @param[in] device  device name, or None to clear
-        @param[in] allowRaise  controls handling an unknown, non-None device:
-            if true raise RuntimeError, else clear the default colorterms
-        """
-        if device is None:
-            Colorterm._activeColorterms = None
+__all__ = ["ColortermNotFoundError", "Colorterm", "ColortermDictConfig", "ColortermLibraryConfig"]
 
-        else:
-            try:
-                Colorterm._activeColorterms = Colorterm._colorterms[device]
-            except KeyError:
-                if allowRaise:
-                    raise RuntimeError("No colour terms are available for %s" % device)
+class ColortermNotFoundError(LookupError):
+    """Exception class indicating we couldn't find a colorterm
+    """
+    pass
 
-                Colorterm._activeColorterms = None
 
-    @staticmethod
-    def getColorterm(band):
-        """!Return the Colorterm for the specified band (or None if unknown)
+class Colorterm(Config):
+    """!Configuration describing a Colorterm
+    
+    The transformed magnitude p' is given by
+        p' = primary + c0 + c1*(primary - secondary) + c2*(primary - secondary)**2
+    """
+    primary = Field(dtype=str, doc="name of primary filter")
+    secondary = Field(dtype=str, doc="name of secondary filter")
+    c0 = Field(dtype=float, default=0.0, doc="Constant parameter")
+    c1 = Field(dtype=float, default=0.0, doc="First-order parameter")
+    c2 = Field(dtype=float, default=0.0, doc="Second-order parameter")
+
+    # the following is desired to allow positional data, but is impossible due to DM-2381
+    # def __init__(self, primary, secondary, c0=0.0, c1=0.0, c2=0.0):
+    #     """!Construct a Colorterm
+
+    #     This overrides the default constructor for improved error detection
+    #     and to allow positional arguments
+    #     """
+    #     Config.__init__(self, primary=primary, secondary=secondary, c0=c0, c1=c1, c2=c2)
```

Comment:
Now that you have a workaround form DM-2831, can you re-enable this?  If not, probably best to just delete it.